share/doc/homebrew/Formula-Cookbook.md

```diff
@@ -345,79 +275,60 @@ Thus, if you change the name of the class, you must also rename the file. Filena
+The preferred commit message format for simple version updates is `foobar 7.3` and for fixes is `foobar: fix flibble matrix.`.
 
+Ensure you reference any relevant GitHub issue e.g. `Closes #12345` in the commit message. Homebrew’s history is the first thing future contributors will look to when trying to understand the current state of formulae they’re interested in.
 
 ## Push
 
-Now you just need to push back to GitHub.
+Now you just need to push your commit to GitHub.
 
 If you haven’t forked Homebrew yet, [go to the repo and hit the fork button](https://github.com/Homebrew/homebrew).
 
 If you have already forked Homebrew on GitHub, then you can manually push (just make sure you have been pulling from the Homebrew/homebrew master):
 
-    git push git@github.com:myname/homebrew.git <what-you-called-your-branch>
+```shell
+git push https://github.com/myname/homebrew/ <what-you-called-your-branch>
+```
 
-Now, please open a Pull Request (on your GitHub repo page) for new and updated brews.
+Now, please [open a pull request](https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged) for your changes.
 
 *   One formula per commit; one commit per formula
-*   Keep merge commits out of the request
-*   If you have any merge or mixup commits, please [squash](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html) them.
-
-If a commit touches multiple files, or isn’t one logical bug fix, or a file is touched in multiple commits, we’ll probably ask you to `rebase` and `squash` your commits. For this reason, you should avoid pushing to your `master` branch. Note, after rebase and/or squash, you'll need to push with `--force` to your remote.
-
-
-# Overview of the Formula Install Process
-<!-- TODO rewrite this. It is outdated, there are more layers now, and the implementation details discussed here are not relevant or useful to most users anyway. -->
-
-*   The result of `Formula.download_strategy` is instantiated.
-*   `DownloadStrategy.fetch` is called (downloads tarball, checks out git repository, etc.)
-*   A temporary sandbox is created in `/tmp/$formulaname`
-*   `DownloadStrategy.stage` is called (extracts tarball to above sandbox, exports git repository to sandbox, etc.)
-*   Patches are applied
-*   Current directory is changed to the stage root (so when you `system make`, it works)
-*   `Formula.install` is called
-*   Anything installed to the keg is cleaned (see later)
-*   The keg is symlinked into Homebrew’s prefix
-*   Caveats are displayed
-
+*   Keep merge commits out of the pull request
 
 # Convenience Tools
-
 ## Messaging
 
 Three commands are provided for displaying informational messages to the user:
```

Comment:
As discussed elsewhere, let's replace `onoe` with `odie`:

> Why should anyone use `onoe`? Right now the only difference is that it's printed to `stderr` and has a different color. Maybe `onoe` should be removed from the documentation and replaced with `odie`, also noting that it will abort the installation.

The example underneath should be also removed or replaced with a better one. Maybe something like:

```ruby
if build.with?("qt") && build.with("qt5")
  odie "Options --with-qt and --with-qt5 are mutually exclusive."
end
system "make", "install"
```